platform/linux-generic/odp_timer.c

```diff
@@ -824,35 +818,26 @@ static odp_event_t timer_cancel(timer_pool_t *tp, uint32_t idx)
 	odp_event_t old_event;
 
 #if USE_128BIT_ATOMICS
-	tick_buf_t new, old;
+	odp_u128_t new, old;
```

Comment:
To avoid casts below, it could be better to insert u128 as a union member in tick_buf_t.